### PR TITLE
Added features

### PR DESCRIPTION
**Pull request by Jkayed**
### What has been done?

- Added Nav Bar with Links
- Added Search feature
- Added Saved Page
- Added Dashboard page
- Added Dashboard Sidebar
- Added All Cryptos page
- Added Top cryptos Page
- Added Categories Page
- Added Search Page
- Downloaded Chart.js Library

### How can we use these new features?
Use the nav bar to navigate to the dashboard page and use the left side bar on the dashboard page to navigate.

### Notes
There is still work that needs to be done on the nav bar and dashboard page, these final touches will be added throughout the week and I'll make PRs as I go.

### Comments/Questions
If you have any questions please let me know!

### CODE REVIEW

File: client/src/Components/CategoriesList.jsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            <th>Top Gainer</th>
+            <th>Volume</th>
+            <th>23hr volume</th>
+            <th>Top 3</th>
+          </tr>
+        </thead>
+        <tbody>
+          {categories.map((crypto) => {
+            return (
+              <tr>
```
Comment (keldim) [inline client/src/Components/CategoriesList.jsx]: could you change this to `<tr key={crypto.id}>` ?
it seems like react requires you to have a key attribute

File: client/src/Components/TopCryptos.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          <tr>
+            <th>Name</th>
+            <th>Price</th>
+            <th>Rank</th>
+          </tr>
+        </thead>
+        <tbody>
+          {topCryptos.map((crypto) => {
+            return (
+              <tr>
```
Comment (keldim) [inline client/src/Components/TopCryptos.jsx]: could you change this to `<tr key={crypto.id}>` ?
it seems like react requires you to have a key attribute

File: client/src/Components/CryptoList.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            <th>Name</th>
+            <th>Price</th>
+            <th>Cap</th>
+            <th>24hr high</th>
+          </tr>
+        </thead>
+        <tbody>
+          {cryptos.map((crypto) => {
+            return (
+              <tr>
```
Comment (keldim) [inline client/src/Components/CryptoList.jsx]: could you change this to `<tr key={crypto.id}>` ?
it seems like react requires you to have a key attribute

**Review by Jkayed — COMMENTED** [review]
Added the Keys

File: client/src/Components/Search/SearchPage.jsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  console.log(cryptoInfo);
+  console.log(search);
+  return (
+    <div>
+      <div
+        className="avatar"
+        style={{ width: 170, height: "auto", marginLeft: 50, marginTop: 20 }}
+      >
+        <img
+          src={cryptoInfo.length === undefined && cryptoInfo.image.thumb}
```
Comment (keldim) [inline client/src/Components/Search/SearchPage.jsx]: could you change this to `src={cryptoInfo.length === undefined ? cryptoInfo.image.thumb : undefined}` ?
error message in the console recommends to write this way

File: client/src/Components/Search/SearchPageTrade.jsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+          cryptoInfo.length === undefined && cryptoInfo.links.blockchain_site[0]
+        }
+      >
+        <span className="exchange">Exhange:</span>
+        <span className="exchange-value">
+          1 {cryptoInfo.length === undefined && cryptoInfo.symbol} = $
+          {cryptoInfo.length === undefined && cryptoInfo.market_data.current_price.usd}
+        </span>
+        <button
+          class="btn btn-xs sm:btn-sm md:btn-md lg:btn-lg"
```
Comment (keldim) [inline client/src/Components/Search/SearchPageTrade.jsx]: could you put these into className? react says to put these into className

File: client/src/Components/Search/SearchPageTrade.jsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        marginTop: -702,
+      }}
+    >
+      <h2 className="oneCryptoValue">
+        $
+        {cryptoInfo.length === undefined &&
+          cryptoInfo.market_data.current_price.usd}
+      </h2>
+      <a
+        href={
```
Comment (keldim) [inline client/src/Components/Search/SearchPageTrade.jsx:25]: could you change this to `href={cryptoInfo.length === undefined ? cryptoInfo.links.blockchain_site[0] : undefined}` ?
error message in the console recommends to write this way

File: client/src/Components/Search/SearchPageTrade.jsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+          </h2>
+        </div>
+        <div className="grid h-20 card bg-base-300 rounded-box place-items-center">
+          <CgArrowsExchangeAltV className="exchange-icon" />
+          <h2 style={{ display: "flex" }}>
+            <span className="get-value">You get</span>
+            <span>
+              <img
+                className="Get-Logo"
+                src={cryptoInfo.length === undefined && cryptoInfo.image.thumb}
```
Comment (keldim) [inline client/src/Components/Search/SearchPageTrade.jsx]: could you change this to `src={cryptoInfo.length === undefined ? cryptoInfo.image.thumb : undefined}` ?
error message in the console recommends to write this way

File: client/src/Components/Search/SearchPageNews.jsx [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+          overflowX: "hidden",
+          overflowY: "auto",
+        }}
+      >
+        <h2 className="News-title">News</h2>
+        <p className="Trending">Trending</p>
+        {cryptoNews.length > 0 &&
+          cryptoNews.map((news) => {
+            return (
+              <div>
```
Comment (keldim) [inline client/src/Components/Search/SearchPageNews.jsx]: could you change this to `<div key={news.id}>` ?
it seems like react requires you to have a key attribute
if there is no id for news, then I believe you can use anything that is unique to each news item in the key attribute

File: client/src/Components/Search/SearchPageRecentTransactions.jsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+          height: 400,
+          marginLeft: 50,
+          marginTop: 450,
+          borderRadius: 10,
+          overflowX: "hidden",
+          overflowY: "auto",
+        }}
+      >
+        {cryptoInfo.length === undefined && cryptoInfo.tickers.map((crypto) => {
+        return <div>
```
Comment (keldim) [inline client/src/Components/Search/SearchPageRecentTransactions.jsx]: could you change this to `<div key={crypto.id}>` ?
it seems like react requires you to have a key attribute

**Review by keldim — CHANGES_REQUESTED** [review]

File: client/src/Components/CategoriesList.jsx [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+                  >
+                    details
+                  </button>
+                </th>
+              </tr>
+            );
+          })}
+        </tbody>
+        {/* foot */}
+        <tfoot>
```
Comment (keldim) [inline client/src/Components/CategoriesList.jsx:111]: hey jamal, is this row for a separate table? would it be okay to delete this?

File: client/src/Components/CryptoList.jsx [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+                  >
+                    details
+                  </button>
+                </th>
+              </tr>
+            );
+          })}
+        </tbody>
+        <tfoot>
+          <tr>
```
Comment (keldim) [inline client/src/Components/CryptoList.jsx:77]: hey jamal, is this row for a separate table? would it be okay to delete this?

**Pull request closed without merging**